contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -78,6 +78,37 @@ func (h *htlcIncomingContestResolver) processFinalHtlcFail() error {
 	return nil
 }
 
+// Launch will call the inner resolver's launch method if the preimage can be
+// found, otherwise it's a no-op.
+func (h *htlcIncomingContestResolver) Launch() error {
+	// NOTE: we don't mark this resolver as launched as the inner resolver
+	// will set it when it's launched.
+	if h.isLaunched() {
+		h.log.Tracef("already launched")
+		return nil
+	}
```

Comment:
I did not find where we ever set the launched variable to true for the two contest resolvers 

was looking for `markLaunched()`

Reply:
it embeds the success resolver as stated in the above comments `NOTE: ...`